### ISSUE TITLE
Shows test passing when there's an error

### ISSUE THREAD

**Issue by nobuttsseb** [cypress-io/cypress#2720]
### Current behavior:

Cypress is looking for an element then timesout when it cannot find it. It does not show that the test has failed, it is green (last test/assertion/'Should fit the image into the canvas').

![image](https://user-images.githubusercontent.com/24407870/48007848-58c79000-e110-11e8-8391-aeb2a501ad72.png)

![image](https://user-images.githubusercontent.com/24407870/48007951-97f5e100-e110-11e8-979a-dbc2043488e3.png)

### Desired behavior:

Should turn the test heading as failed/red when there is a failure to test.

### Steps to reproduce:

- Add a test that looks for a dom element that does not exist.
- Run the test.

### Versions

Cypress: 3.1.0
OS: Mac
Browser: Chrome

**Comment by jennifer-shehane** [comment]
Do you have the assertions in an `after` hook? May be this issue: https://github.com/cypress-io/cypress/issues/2296

**Comment by jennifer-shehane** [comment]
Hey @nobuttsseb, did you have the assertions in an after hook? Or could you provide the test code you wrote for this? We will have to close this otherwise.

**Comment by jennifer-shehane** [comment]
Closing. Cannot reproduce with information provided. Please comment with a reproducible example if you are still experiencing this issue and we can reopen.

**Comment by josh-atkins** [comment]
I received similar behaviour (passing test but failing assertion) where the assertion had an expectation along the lines of `assert expected { Object (length, prevObject, ...) } to equal `, and gave an exception along the lines of `(uncaught exception) AssertionError: Timed out retrying after 4000ms: Expected to find element: YOUR_ELEMENT, but never found it.`

Similar to what HamzeMurtaza noted above, the cause for me was leaving an `async` function definition in the `it` block, but not following through and with `await` on anything in the `it` body. 

In other words, this will show a passing test but fail on assertions:
```
    it('shows passing test but fails assertions', async () => {
        // test body, but no `await`
    })
```

But removing the async or adding an await function within the body will cause the test to run as expected.
